articles/react-native-firebase-facebook-authentication/index.md

```diff
@@ -0,0 +1,451 @@
+    implementation 'com.google.firebase:firebase-auth'
+}
+```
+
+Next, let's install the React Native Facebook SDK.
+
+```bash
+npm install react-native-fbsdk
+```
+
+In `android/build.gradle`, add `mavenCentral()` under `repositories` in `buildscript`:
+
+```gradle
+buildscript {
+    repositories {
+        google()
+        jcenter()
+
+        // Add this line
+        mavenCentral()
+    }
+}
+```
+
+In `android/app/build.gradle`, add the following under `dependencies`:
+
+```gradle
+implementation 'com.facebook.android:facebook-android-sdk:[5,6)'
+```
+
+Now, you should add your Facebook App ID to your project's strings file in `android/app/res/values/strings.xml`.
+
+```XML
+<resources>
+  <string name="app_name">RNFirebaseFacebookAuth</string>
+  <string name="facebook_app_id"><-- Your App ID here --></string>
+</resources>
+```
+
+In the `AndroidManifest.xml` file, add a `uses-permission` element under `manifest`: (Ignore if already present)
+
+```XML
+<uses-permission android:name="android.permission.INTERNET" />
+```
+
+Next, add a `meta-data` to the `application` element:
+
+```XML
+<application android:label="@string/app_name" ...>
+    ...
+    <meta-data android:name="com.facebook.sdk.ApplicationId" android:value="@string/facebook_app_id"/>
+    ...
+</application>
+```
+
+Follow this [Getting Started Guide](https://developers.facebook.com/docs/ios/getting-started/?sdk=cocoapods) to set up the SDK for iOS.
+
+### Sign-in
+
+In *App.js*, Let's create a function called `signIn` which will be called when the user presses the Sign in button. Let's wrap all the code inside the function in a `try/catch` block to display any error that may arise during the sign in flow.
+
+```JSX
+async function signIn() {
+  try {
+
+    // Code goes here
+
+  } catch (error) {
+    alert(error);
+  }
+}
+```
+
+Let's use the Login Manager to log in with permissions. We need to pass an array of permissions that the app needs to the function.
+
+```JSX
+const result = await LoginManager.logInWithPermissions([
+  'public_profile',
+  'email',
+]);
+```
+
+If the result contains an `isCancelled` property, it means the user cancelled the process. Let's throw an error which will then be handled by the `catch` block.
+
+```JSX
+if (result.isCancelled) {
+  throw 'User cancelled the login process';
+}
+```
+
+Once signed in, we should get the users AccessToken.
+
+```JSX
+const data = await AccessToken.getCurrentAccessToken();
+```
+
+If the data is empty, let's throw an error.
+
+```JSX
+if (!data) {
+  throw 'Something went wrong obtaining access token';
+}
+```
+
+Now, we should create a credential using the access token and sign in the user into the app.
+
+```JSX
+const facebookCredential = auth.FacebookAuthProvider.credential(data.accessToken);
+return auth().signInWithCredential(facebookCredential);
+```
+
+Here is the complete code for the `signIn` function:
+
+```JSX
+async function signIn() {
```

Comment:
Since this is a large codeblock, please include code comments so that readers can easily follow.

Reply:
added the comments.